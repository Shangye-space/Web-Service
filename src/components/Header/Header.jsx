import React from 'react';

import styles from './Header.pcss';

const Header = () => {
    return (
        <>
            <div className={styles.top}>This is header my friend</div>
        </>
    );
};

export default Header;
