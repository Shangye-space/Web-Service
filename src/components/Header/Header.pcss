.top {
    height: 40px;
    width: 100%;

    background-color: green;
}

body {
    background-color: black;
}
