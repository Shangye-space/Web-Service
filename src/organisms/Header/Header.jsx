import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import classnames from 'classnames';
import styles from './Header.module.scss';

import UserBar from '../../molecules/UserBar';

const Header = ({ userData }) => {
    return (
        <>
            <nav
                className={classnames(
                    'navbar',
                    'navbar-expand-lg',
                    'navbar-dark',
                    styles.nav,
                )}
            >
                <button
                    className={classnames(
                        'navbar-toggler',
                        styles.toggleButton,
                    )}
                    type="button"
                    data-toggle="collapse"
                    data-target="#navbarTogglerDemo01"
                    aria-controls="navbarTogglerDemo01"
                    aria-expanded="false"
                    aria-label="Toggle navigation"
                >
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div
                    className={classnames(
                        'collapse',
                        'navbar-collapse',
                        styles.nav_wrapper,
                    )}
                    id="navbarTogglerDemo01"
                >
                    <Link className="navbar-brand" to="/">
                        <strong className={styles.orangeText}>Shangye</strong>
                        .space
                    </Link>
                    <ul
                        className={classnames(
                            styles.list,
                            'navbar-nav',
                            'mt-2 mt-lg-0',
                        )}
                    >
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/" exact>
                                Home
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/shop">
                                Shop collection
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/story">
                                Our Story
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink className="nav-link" to="/contacts">
                                Contacts
                            </NavLink>
                        </li>
                    </ul>
                    <UserBar className={styles.acc_bar} userData={userData} />
                </div>
            </nav>
        </>
    );
};

export default Header;
