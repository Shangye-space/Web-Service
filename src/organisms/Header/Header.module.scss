@import '../../settings/variables.scss';

.nav {
    background-color: $default-dark;
    font-family: $marker;
    position: fixed !important;
    top: 0;
    left: 0;
    z-index: 100;
    width: 100%;

    .acc_bar {
        margin-left: 16px;
        flex-flow: nowrap;
    }
}

.nav_wrapper {
    justify-content: space-between;
    padding: 2px 2rem !important;
}

.toggleButton {
    margin-left: auto;
}

.orangeText {
    color: $soft-orange;
}

.list {
    display: flex;
    min-width: 448px;
    max-width: calc(100% - 800px);
    justify-content: space-between;
}
