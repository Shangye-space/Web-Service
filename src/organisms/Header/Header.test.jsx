import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from './Header.jsx';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Router>
            <Header userData={{ name: 'Test' }} />
        </Router>,
        div,
    );
    ReactDOM.unmountComponentAtNode(div);
});

it('snapshot renders', () => {
    const component = renderer.create(
        <Router>
            <Header userData={{ name: 'Test' }} />
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
