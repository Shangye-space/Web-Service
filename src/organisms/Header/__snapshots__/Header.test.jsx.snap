// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark nav"
>
  <button
    aria-controls="navbarTogglerDemo01"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler toggleButton"
    data-target="#navbarTogglerDemo01"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse nav_wrapper"
    id="navbarTogglerDemo01"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <strong
        className="orangeText"
      >
        Shangye
      </strong>
      .space
    </a>
    <ul
      className="list navbar-nav mt-2 mt-lg-0"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/shop"
          onClick={[Function]}
        >
          Shop collection
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/story"
          onClick={[Function]}
        >
          Our Story
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-link"
          href="/contacts"
          onClick={[Function]}
        >
          Contacts
        </a>
      </li>
    </ul>
    <div
      className="userBarWrapper"
    >
      <div
        className="cartWrapper"
      >
        <a
          className="fas fa-shopping-cart"
        />
      </div>
      <div
        className="accountWrapper"
        onClick={[Function]}
      >
        <div
          className="fas fa-user-circle"
        />
        <span>
          Hi, 
          Test
          !
        </span>
      </div>
    </div>
  </div>
</nav>
`;
